### ISSUE TITLE
Ubuntu gnome-session 100% CPU utilization

### ISSUE THREAD

**Issue by ppenzin** [DC-DeepComputing/DC-ROMA_Pad_II_RV-P1B#1]
DC Roma Pad II 16G RAM / 128G EMMC, after updating software and rebooting `gnome-session` goes to 100% CPU utilization making the device basically unusable (to even see processes that you have to actually log into it remotely). Any help or suggestions would be greatly appreciated.

**Comment by ppenzin** [comment]
Tried disabling Wayland - that didn't help, instead it disabled all direct screen and keyboard inputs.

**Comment by tyokaisan** [comment]
in my case ,
i have done  ' apt update ; apt upgrade '  
 then DC Roma Pad II 16G RAM / 128G EMMC become basically unusable .

so i want to the original state .
what is ' download key  (beneath volume down )'  ? , 
and  how to use it ?

**Comment by mlang99** [comment]
I also did the update and observed the same performance reduction issue.

Same question: what does the download key do?

**Comment by tyokaisan** [comment]
please look at https://github.com/DC-DeepComputing/DC-ROMA_Pad_II_RV-P1B/blob/main/DC-ROMA_Pad_II_Manual.pdf .

the left photo shows  'DOWN LOAD KEY ' .
you also know the debian image in 32GB micro SD card in the package .

so i suppose that the use of  'DOWN LOAD KEY ' may brings DC-ROMA_Pad_II to original state , but this is my imagination .

**Comment by DC-DeepComputing** [comment]
The download key function: Press and hold the download key to power on, enter the flashing mode, and then you can connect with the Type-C interface for flashing.  
BTW, currently the preinstalled Ubuntu desktop 24.04 with PAD II don't support directly upgade by system due to there may be potiential conflicts for some drivers or libs. We are continually working on improve the support of RISC-V for ubuntu desktop 24.04 and will publish the new version of it later. 
If you alreade upgraded, you can use debian images in TF cards to reset the os to debian. If you more prefer ubuntu os,  you can mail us to get Ubuntu desktop 23.10 for the pad.

**Comment by tyokaisan** [comment]
I still want ubuntu for my pad Ⅱ.
I have one question. If I insert a 32GB SD card into the TF card slot and
then turn it on, will it go back to the original ubuntu?

I will pay by credit card, just like when I bought the pad Ⅱ, but I'm also
fine with cash on delivery.

2024年10月21日(月) 15:16 DC-DeepComputing ***@***.***>:

**Comment by DC-DeepComputing** [comment]
Yes, if you burn the image to the TF card and insert the card to the TF card slot, then to turn it on, the pad will boot from the TF card.  
You need not pay for ubuntu desktop image used by DC-ROMA pad ii. We will provide the image to our PAD customer. Please mail to sales@deepcomputing.io with your order number and we will sentd you the image download link.

**Comment by DoktorCranium** [comment]
Hello - how do you flash the Debian image from the TF card to the internal EMMC disk ? Is it possible to do directly off the booted TF card, or is there some other procedure. 
Thanks

**Comment by tyokaisan** [comment]
i use ubuntu-24.04-desktop-k1-9985.img .

but
When I put in the micro SD card and turned it on,
it did start up, but it stopped halfway through

so I couldn't put it back to original gnome .

**Comment by DoktorCranium** [comment]
My SD card just boots into Debian 12 live image (for user roma) and runs off fully from SD card.  Do you need to hold the download pin on the device while booting off the SD card to flash this to the EMMC ?

**Comment by DC-DeepComputing** [comment]
> My SD card just boots into Debian 12 live image (for user roma) and runs off fully from SD card. Do you need to hold the download pin on the device while booting off the SD card to flash this to the EMMC ?

No, the Debian images provided in the package is used to boot from SD. If you hope to burn the images to EMMC of PAD, please mail to sales@deepcomputing.io for related images. However, we currently has only Ubuntu desktop 23.10 and 24.04 images which can be burned into the EMMC.

**Comment by DoktorCranium** [comment]
Thank you

**Comment by DC-DeepComputing** [comment]
Here is a video from our youtube channel to demo how to reset the factory image of DC-ROMApad II . Please refer the steps if you want to do so. [DeepComputing Youtube channel](https://www.youtube.com/watch?v=EOOFwBAWPJc)

**Comment by tyokaisan** [comment]
thanks to your good advices ,
 my laptop return to the original state .

now i realized that ' apt update ' is very dangerou now .

many thanks !

2024年10月25日(金) 19:50 DC-DeepComputing ***@***.***>:

**Comment by Haui1112** [comment]
thank you for providing this. it helped me greatly. can you let us know which packages are safe to update and which are not? a lot of software installation fails because of broken packages in ubuntu. other stuff I dont want to install because it triggers an update to the packages.

**Comment by Mabin-J** [comment]
I found to fix it.

---
1. Press `Ctrl + Alt + F2` in login screen
2. login
3. run `sudo su - ` and input password and press `enter`
4. rename `/etc/apt/sources.list.d/bianbu.sources.1` to `/etc/apt/sources.list.d/bianbu.sources`
5. modify `bianbu.sources`
    - URIs to `http://archive.spacemit.com/bianbu/`
    - result `bianbu.sources`'s content
```
Types: deb
URIs: http://archive.spacemit.com/bianbu/
Suites: noble-porting/snapshots/v2.0beta1
Components: main universe restricted multiverse
Signed-By: /usr/share/keyrings/bianbu-archive-keyring-noble.gpg
```
6. run `apt update` and `apt dist-upgrade`

---
```
sudo su - 
mv /etc/apt/sources.list.d/bianbu.sources.1 /etc/apt/sources.list.d/bianbu.sources
sed "s/URIs:.*/URIs: http:\/\/archive.spacemit.com\/bianbu\//g" -i /etc/apt/sources.list.d/bianbu.sources
apt update
apt dist-upgrade
```

**Comment by DoktorCranium** [comment]
The upgrade process updates the core mesa packages which then cause the CPU spike due to software rendering.
So as long as you dont touch these you should be fine I believe.

**Comment by Mabin-J** [comment]
next my setting
```
sudo su - 
mv /etc/apt/sources.list.d/bianbu.sources.1 /etc/apt/sources.list.d/bianbu.sources
sed "s/URIs:.*/URIs: http:\/\/archive.spacemit.com\/bianbu\//g" -i /etc/apt/sources.list.d/bianbu.sources
sed "s/Suites:.*/Suites: noble-porting\/snapshots\/v2.0.1/g" -i /etc/apt/sources.list.d/bianbu.sources
apt-mark hold linux-image-6.6.36
apt-mark hold u-boot-spacemit
apt update
apt dist-upgrade
```

**Comment by alefnode** [comment]
If you need the link (showed in youtube) to the image: http://120.92.155.32:8082/ui/native/virtOS/k1-107/ubuntu-24.04-desktop-k1-pad-9985-upgradecard.zip

Thanks for the tips about the sources.list my problems is for me it is saying me timeout every time i try to dist-upgrade

**Comment by ppenzin** [comment]
The Ubuntu images didn't work for me for some reason, screen just stays blank, used `dd` to burn like so (same for 23.10):

```
$ sudo dd if=ubuntu-24.04-desktop-k1-pad-9985-upgradecard.img  of=/dev/mmcsd0 bs=1M conv=sync status=progress
```

Edit: same when using Balena on Mac.

Debian boots just fine, but that is a live image, unfortunately. When I take the card out, the original install still boots. I'm going to try to apply the recommendations above in a remote shell.

**Comment by Haui1112** [comment]
Can you try using only dd with if and of. without the parameters and report back if that helped? This is what worked for me.

**Comment by Haui1112** [comment]
as mentioned in #4, you can do ```sudo unminimize``` and the system will update without breaking gnome. But beware: i have not attempted another update after that. the mesa drivers are still old in my understanding.

**Comment by ppenzin** [comment]
Long story short is that dependencies are broken, should we close this? @DC-DeepComputing can we create a wiki or a maybe a readme file we can all contribute to?

**Comment by Haui1112** [comment]
> sudo su - 
mv /etc/apt/sources.list.d/bianbu.sources.1 /etc/apt/sources.list.d/bianbu.sources
sed "s/URIs:.*/URIs: http:\/\/archive.spacemit.com\/bianbu\//g" -i /etc/apt/sources.list.d/bianbu.sources
sed "s/Suites:.*/Suites: noble-porting\/snapshots\/v2.0.1/g" -i /etc/apt/sources.list.d/bianbu.sources
apt-mark hold linux-image-6.6.36
apt-mark hold u-boot-spacemit
apt update
apt dist-upgrade

So, upgrading with the spacemit bianbu sources in place did not work without errors, although I held on to the packages for uboot and the kernel. here is a paste of the output: https://paste.giftedmc.com/?f01f46ec4f2e5354#CqDFEuSwixDahRVFb5Vf6yhZdTQ7tTRfQQkqmHV5z1Q7

Edit-Summary: A lot of stuff from the official ubuntu repo got "downgraded" to the bianbu version, sometimes literally (to a lower release) and it ran into errors at the end.

BUT the 100% CPU core usage is gone now.

**Comment by Mabin-J** [comment]
I think that re-run `sudo apt dist-upgrade` can fix it. If not, reinstall `k1x-cam` can fix it.
```
sudo apt remove k1x-cam-lib
sudo apt install k1x-cam
```

And, I think that showed `downgrade` maybe not downgrade because version naming rule is different. But I don't know.

**Comment by Haui1112** [comment]
an additional dist-upgrade fixed it. Thank you!

**Comment by ppenzin** [comment]
Catching up on the thread after some time, tried latest suggestions:

- Additional `dist-upgrade` after `apt-mark hold`
- Reinstall `k1x-cam`

Both report success, but no luck so far, still 100% CPU utilization. In #10 it looks like @DoktorCranium and myself have roughly the same device but observe different behavior when we try to reimage.

**Comment by ppenzin** [comment]
> The download key function: Press and hold the download key to power on, enter the flashing mode, and then you can connect with the Type-C interface for flashing.

@DC-DeepComputing is there an image to recover this way?

**Comment by DC-DeepComputing** [comment]
> > The download key function: Press and hold the download key to power on, enter the flashing mode, and then you can connect with the Type-C interface for flashing.
> 
> @DC-DeepComputing is there an image to recover this way?

the download key here designed for upgrading android aosp in future.
We mainly provide  ubuntu 24.04 upgare images to burn the emmc and recovere to factory image.  However,seems you have failed on that.

**Comment by mlang99** [comment]
I found a new version on JFrog:
http://120.92.155.32:8082/artifactory/virtOS/k1-107/ubuntu-24.04-desktop-k1-pad-10814-v2.0.1-upgradecard.zip

I installed this and successfully updated without the 100% cpu usage issue. Works well so far.

**Comment by DC-DeepComputing** [comment]
> I found a new version on JFrog: http://120.92.155.32:8082/artifactory/virtOS/k1-107/ubuntu-24.04-desktop-k1-pad-10814-v2.0.1-upgradecard.zip
> 
> I installed this and successfully updated without the 100% cpu usage issue. Works well so far.

Thanks for you feedback. This is for the latest ubuntu desktop 24.04 version for the PAD. Here is also another download link of the same version ： https://deepcomputing.io/support/download/os/ubuntu-24.04-desktop-k1-pad-10814-v2.0.1-upgradecard.zip
